addons/web/static/src/js/public/public_crash_manager.js

```diff
@@ -0,0 +1,65 @@
+odoo.define('web.public.CrashManager', function (require) {
+"use strict";
+
+var core = require('web.core');
+var CrashManager = require('web.CrashManager').CrashManager;
+var mixins = require('web.mixins');
+var ServicesMixin = require('web.ServicesMixin');
+var ServiceProviderMixin = require('web.ServiceProviderMixin');
+
+var _t = core._t;
+
+var PublicCrashManager = CrashManager.extend(mixins.EventDispatcherMixin, ServicesMixin, ServiceProviderMixin, {
+    init: function () {
+        mixins.EventDispatcherMixin.init.call(this);
+        this._super.apply(this, arguments);
+        ServiceProviderMixin.init.call(this);
+    },
+
+    notifyConnectionState: function () {
```

Comment:
It is in ```abstract_web_client.js``` with the ```_onConnectionRestored``` and the ```_onConnectionLost``` method.
Now that front and back notification system are reconcile maybe it would be a good thing to use the same one.
We should discuss how to do it !

Reply:
Same as for the window.onerror of your first commit. I don't see the point of making this the responsibility of the web client, move the behavior in the crash manager directly. This can be done in a separate commit (before this one).